app/app-config-schema.json

```diff
@@ -0,0 +1,406 @@
+        "email_time_span",
+        "email_time_span_in_words",
+        "email_subject_name",
+        "sender_email_address",
+        "sendgrid_api_key",
+        "email_list"
+      ]
+    },
+    "slack_config": {
+      "type": "object",
+      "properties": {
+        "slack_channel": {
+          "type": "string"
+        },
+        "slack_hook_url": {
+          "type": "string"
+        },
+        "slack_run_interval": {
+          "type": "integer"
+        },
+        "slack_notification_rules": {
+          "type": "object",
+          "properties": {
+            "category_based_rules": {
+              "type": "object",
+              "properties": {
+                "Notification": {
+                  "type": "array",
+                  "items": {}
+                }
+              },
+              "required": [
+                "Notification"
+              ]
+            },
+            "keyword_based_rules": {
+              "type": "object",
+              "properties": {
+                "crash": {
+                  "type": "array",
+                  "items": {}
+                }
+              },
+              "required": [
+                "crash"
+              ]
+            }
+          },
+          "required": [
+            "category_based_rules",
+            "keyword_based_rules"
+          ]
+        }
+      },
+      "required": [
+        "slack_channel",
+        "slack_hook_url",
+        "slack_run_interval",
+        "slack_notification_rules"
+      ]
+    },
+    "jira_config": {
+      "type": "object",
+      "properties": {
+        "base_url": {
+          "type": "string"
+        },
+        "project_id": {
+          "type": "string"
+        },
+        "story_type": {
+          "type": "string"
+        },
+        "bug_type": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "base_url",
+        "project_id",
+        "story_type",
+        "bug_type"
+      ]
+    },
+    "algorithm_config": {
+      "type": "object",
+      "properties": {
+        "categorization_algorithm": {
+          "enum": ["text_match","lstm_classification"]
+        },
+        "algorithm_days_filter": {
+          "type": "integer"
+        },
+        "bug_feature_keywords_file": {
+          "type": "string"
+        },
+        "bug_feature_keywords_weights_file": {
+          "type": "string"
+        },
+        "category_keywords_file": {
+          "type": "string"
+        },
+        "category_keywords_weights_file": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "categorization_algorithm",
+        "algorithm_days_filter",
+        "bug_feature_keywords_file",
+        "bug_feature_keywords_weights_file",
+        "category_keywords_file",
+        "category_keywords_weights_file"
+      ]
+    },
+    "env_keys": {
+      "type": "array",
+      "items": {
+          "type": "string"
+      }
+    },
+    "review_channels": {
```

Comment:
I think you should create a base review channel object and then derive all the other review channel object from those and finally do a **oneOf** of all the objects.

Reply:
I was under the assumption that it can have more than one review channel, as its an array. I have tried with **oneOf** initially, which validated only the first review channel.

I have added _if_ checks per channel level, with the relevant fields for that channel. is that sufficient?